src/objective-c/tests/ThirdPartyTests/Libuv/LibuvGetAddrInfoTests.m

```diff
@@ -1,141 +1,35 @@
-/*
- *
- * Copyright 2021 gRPC authors.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- *
- */
-
 #import <XCTest/XCTest.h>
 
-#import <stdlib.h>
-#import <uv.h>
-
-#define ASSERT(x) XCTAssertTrue(x)
-#define ASSERT_NULL(x) XCTAssertTrue(x == NULL)
-#define ASSERT_NOT_NULL(x) XCTAssertFalse(x == NULL)
-#define CONCURRENT_COUNT 10
-
-static const char* name = "localhost";
-
-static int getaddrinfo_cbs = 0;
+#import "test/runner.h"
+#import "test/task.h"
 
-/* data used for running multiple calls concurrently */
-static uv_getaddrinfo_t* getaddrinfo_handle;
-static uv_getaddrinfo_t getaddrinfo_handles[CONCURRENT_COUNT];
-static int callback_counts[CONCURRENT_COUNT];
-static int fail_cb_called;
-
-static void getaddrinfo_fail_cb(uv_getaddrinfo_t* req, int status, struct addrinfo* res) {
-  ASSERT(fail_cb_called == 0);
-  ASSERT(status < 0);
-  ASSERT_NULL(res);
-  uv_freeaddrinfo(res); /* Should not crash. */
-  fail_cb_called++;
-}
-
-static void getaddrinfo_basic_cb(uv_getaddrinfo_t* handle, int status, struct addrinfo* res) {
-  ASSERT(handle == getaddrinfo_handle);
-  getaddrinfo_cbs++;
-  free(handle);
-  uv_freeaddrinfo(res);
-}
+// Tests in:
+// libuv/test/test-getaddrinfo.c
 
-static void getaddrinfo_cuncurrent_cb(uv_getaddrinfo_t* handle, int status, struct addrinfo* res) {
-  int i;
-  int* data = (int*)handle->data;
+TEST_DECLARE(getaddrinfo_fail)
+TEST_DECLARE(getaddrinfo_fail_sync)
+TEST_DECLARE(getaddrinfo_basic)
+TEST_DECLARE(getaddrinfo_basic_sync)
+TEST_DECLARE(getaddrinfo_concurrent)
 
-  for (i = 0; i < CONCURRENT_COUNT; i++) {
-    if (&getaddrinfo_handles[i] == handle) {
-      ASSERT(i == *data);
-      callback_counts[i]++;
-      break;
-    }
-  }
-  ASSERT(i < CONCURRENT_COUNT);
-  free(data);
-  uv_freeaddrinfo(res);
-  getaddrinfo_cbs++;
-}
+#define TASK_LIST_START_LOCAL static task_entry_t TEST_TASKS[] = {
+TASK_LIST_START_LOCAL
+TEST_ENTRY(getaddrinfo_fail)
+TEST_ENTRY(getaddrinfo_fail_sync)
+TEST_ENTRY(getaddrinfo_basic)
+TEST_ENTRY(getaddrinfo_basic_sync)
+TEST_ENTRY(getaddrinfo_concurrent)
+TASK_LIST_END
 
-@interface LibuvGetAddrInfoTests : XCTestCase
+@interface LibuvGetAddrInfoTests2 : XCTestCase
```

Comment:
convention is to name test suite the same as test file (LibuvGetAddrInfoTests). Any issue not being able to use LibuvGetAddrInfoTests ?